Source/Managers/DFU/FirmwareUpgradeManager.swift

```diff
@@ -388,17 +423,28 @@ public class FirmwareUpgradeManager : FirmwareUpgradeController, ConnectionObser
             return
         }
         // Check that the image array exists.
-        guard let images = response.images else {
+        guard let responseImages = response.images else {
             self.fail(error: FirmwareUpgradeError.invalidResponse(response))
             return
         }
-        // Check that we have 2 images in the array.
-        if images.count != 2 {
-            self.fail(error: FirmwareUpgradeError.unknown("Test response does not contain enough info."))
-            return
+
+        // Check that we have the correct number of images in the responseImages array.
+        if self.images.count == 1 {
+            // For Image 0, primary and secondary slots are expected.
+            guard responseImages.count == 2 else {
+                self.fail(error: FirmwareUpgradeError.unknown("Test response expected 2 images."))
+                return
+            }
+        } else {
+            // For Image 1, for security reasons primary slot cannot be returned, only secondary.
```

Comment:
I think this should not be hardcoded. It may be implemented any time and we should not rely on it not being returned, and the lib should accept both: the slot present and absent.